src/by/epam/lab/task3/Runner.java

```diff
@@ -0,0 +1,86 @@
+package by.epam.lab.task3;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Scanner;
+
+public class Runner {
+    private static final int FIRST_SCAN_VALUE = 0;
+    private static final int SECOND_SCAN_VALUE = 1;
+    private static final int THIRD_SCAN_VALUE = 2;
+    private static final int SEARCH_NUMBER_VALUE = 5;
+
+    public static void main(String[] args) {
+        Purchase[] purchases = parsePurchaseFromTxt();
+        for (Purchase a : purchases) {
+            System.out.println(a);
+        }
+        calculatePurchasesStatistic(purchases);
+        Arrays.sort(purchases);
+        for (Purchase a : purchases) {
+            System.out.println(a);
+        }
+        Purchase key = new Purchase(SEARCH_NUMBER_VALUE, 0, 0);
+        System.out.println(purchases[Arrays.binarySearch(purchases, key)]);
+    }
+
+    private static Purchase[] parsePurchaseFromTxt() {
+        Purchase[] purchases = new Purchase[]{};
+        String filepath = "src\\in.txt";
+        try (Scanner scan = new Scanner(new FileReader(filepath))) {
+            scan.useLocale(Locale.ENGLISH);
+            if (scan.hasNext() && scan.hasNextInt()) {
```

Comment:
It's too explicite and difficult for understanding. You can better introduce new variables instead of constants, then use while (scanner.hasNext()) construction with scanner.nextInt() inside body and then counter.

using that you wil get rid of delimiter and constants, and explisite if.